src/Gumps/GumpResourceTracker.cpp

```diff
@@ -0,0 +1,329 @@
+                m_itemsDataBoxGui->Add(txt);
+            }
+        }
+    }
+}
+
+void CGumpResourceTracker::DrawItem()
+{
+    for (uint8_t r = 0; r < numRow; r++)
+    {
+        for (uint8_t c = 0; c < numCol; c++)
+        {
+            TrackedItem item = items[r][c];
+            if (item.graphic != 0)
+            {
+                CIndexObjectStatic &sio = g_Index.m_Static[item.graphic];
+                auto spr = (CSprite *)sio.UserData;
+                if (spr != NULL)
+                {
+                    int tileOffsetX = (boxDimension - 18 - (spr->ImageWidth)) / 2;
+                    int tileOffsetY = (boxDimension - (spr->ImageHeight)) / 2;
+                    m_itemsDataBoxGui->Add(new CGUIShader(&g_ColorizerShader, true));
+                    m_itemsDataBoxGui->Add(new CGUITilepic(
+                        item.graphic,
+                        item.color,
+                        (c * boxDimension) + tileOffsetX,
+                        (r * boxDimension) + tileOffsetY));
+                    m_itemsDataBoxGui->Add(new CGUIShader(&g_ColorizerShader, false));
+                }
+            }
+        }
+    }
+}
+
+void CGumpResourceTracker::DrawGrid()
+{
+    for (uint8_t c = 0; c < numCol; c++)
+    {
+        m_gridDataBoxGUI->Add(new CGUILine(
+            (c + 1) * boxDimension, 5, (c + 1) * boxDimension, boxDimension * numRow, 0x900));
+        for (uint8_t r = 0; r < numRow; r++)
+        {
+            m_gridDataBoxGUI->Add(new CGUILine(
+                0, (r + 1) * boxDimension, boxDimension * numCol, (r + 1) * boxDimension, 0x900));
+        }
+    }
+}
+
+void CGumpResourceTracker::DeleteGrid()
+{
+    m_gridDataBoxGUI->Clear();
+}
+
+int CGumpResourceTracker::CountItemBackPack(uint32_t graph)
+{
+    int result = 0;
+    CGameItem *bk = (CGameItem *)g_Player->FindLayer(OL_BACKPACK);
+    if (bk != nullptr)
+    {
+        for (int i = 0; i < bk->GetItemsCount(); i++)
+        {
+            CGameItem *item = (CGameItem *)bk->Get(i);
+            if (item->IsContainer())
+            {
+                for (int j = 0; j < item->GetItemsCount(); j++)
+                {
+                    CGameItem *subitem = (CGameItem *)item->Get(j);
+                    if (subitem->Graphic == graph)
+                    {
+                        result += subitem->Count;
+                    }
+                }
+            }
+            else if (item->Graphic == graph)
+            {
+                result += item->Count;
+            }
+        }
+    }
+    return result;
+}
+
+void CGumpResourceTracker::UpdateSize()
+{
+    Width = m_StartResizeWidth + g_MouseManager.LeftDroppedOffset().X;
+    Height = m_StartResizeHeight + g_MouseManager.LeftDroppedOffset().Y;
+
+    if (Height < boxDimension)
+    {
+        Height = boxDimension;
+    }
+
+    if (Width < boxDimension)
+    {
+        Width = boxDimension;
+    }
+
+    int maxHeight = 1200;
+    int maxWidth = 1200;
```

Comment:
avoid using magic numbers. make these constants in the class definition, based on the grid size, so:
```
constexpr int BOX_SIZE = 60;
constexpr int GRID_SIZE = 20;
constexpr int MAX_WIDTH = GRID_SIZE * BOX_SIZE;
constexpr int MAX_HEIGHT = GRID_SIZE * BOX_SIZE;
```